vendor/engines/secure_rooms/app/support/secure_rooms/auto_orphan_occupancy.rb

```diff
@@ -0,0 +1,36 @@
+module SecureRooms
+
+  class AutoOrphanOccupancy
+
+    def perform
+      occupancy_order_details.each do |od|
+        od.transaction do
+          orphan_occupancy(od)
+        end
+      end
+    end
+
+    private
+
+    def occupancy_order_details
+      OrderDetail.joins(:occupancy).merge(long_running_occupancies)
+    end
+
+    def long_running_occupancies
+      SecureRooms::Occupancy.current.where(
+        SecureRooms::Occupancy.arel_table[:entry_at].lt(Time.current - 12.hours),
```

Comment:
yeahh this line is 🤢  if you're cool with the readable version I'd go for it too